R/encoloc.R

```diff
@@ -112,7 +133,7 @@ calculate_purity <- function(variants, ext_ld, squared) {
 }
 
 # Main processing function
-process_coloc_results <- function(coloc_result, LD_meta_file_path,analysis_script_obj, PPH4_thres = 0.8, coloc_pip_thres = 0.95, squared = FALSE, min_abs_corr = 0.5, null_index = 0, coloc_index = "PP.H4.abf", analysis_region) {
+process_coloc_results <- function(coloc_result, LD_meta_file_path,analysis_script_obj, PPH4_thres = 0.8, coloc_pip_thres = 0.95, squared = FALSE, min_abs_corr = 0.5, null_index = 0, coloc_index = "PP.H4.abf", analysis_region, median_abs_corr = NULL) {
```

Comment:
this is okay for the package so nothing to change it here setting it to NULL by default. in practice when we use this function, we can consider using median_abs_corr the same as what i did in `susie_twas` pipeline inside of cis_workhorse?

Reply:
Actually you will introduce a bug in:

```
is_pure = which(purity[,1] >= min_abs_corr | purity[,2] >= median_abs_corr)
```

this line will not work if median_abs_corr is NULL